lib/inferno/dsl/fhir_evaluator/evaluator.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module FhirEvaluator
+  class Evaluator
+    def initialize(the_ig)
+      @ig = the_ig
+    end
+
+    def evaluate(data, summary = nil, config = Config.new)
+      summary ||= DataSummary.new(data)
+      context = EvaluationContext.new(@ig, data, summary, config)
+
+      active_rules = []
+      config.data['Rule'].each_key do |rulename|
+        active_rules << rulename if config.data['Rule'][rulename]['Enabled']
+      end
+
+      Rule.descendants.each do |rule|
+        rule.new.check(context) if active_rules.include?(rule.name.gsub('FhirEvaluator::Rules::', ''))
```

Comment:
Assuming the rules are flat and not nested under further namespaces:
```suggestion
        rule.new.check(context) if active_rules.include?(rule.name.demodulize)
```
Even better would be:
```
Rule
  .descendants
  .select { |rule| active_rules.include? ... }
  .each { |rule| rule.new.check ... }
```

Anytime you're using `each` you should consider whether it could be broken up into combinations of `select`, `map`, `reduce`, etc.

Reply:
Done

Reply:
I don't see any changes here.